### PR TITLE
LightningEntity mappings

### PR DESCRIPTION
**Pull request by MisterPeModder**

### CODE REVIEW

File: mappings/net/minecraft/entity/LightningEntity.mapping [hunk 1]
```diff
@@ -1 +1,16 @@
 CLASS app net/minecraft/entity/LightningEntity
+	FIELD b randomSeed J
+	FIELD c ticksBeforeDespawn I
+	FIELD d fireSpawnCount I
+	FIELD e preventDamage Z
+	FIELD f causingPlayer Luv;
```
Comment (ChloeDawn) [inline mappings/net/minecraft/entity/LightningEntity.mapping]: `channeller`

File: mappings/net/minecraft/entity/LightningEntity.mapping [hunk 2]
```diff
@@ -1 +1,16 @@
...
+	METHOD <init> (Lbdb;DDDZ)V
+		ARG 1 world
+		ARG 2 x
+		ARG 4 y
+		ARG 6 z
+		ARG 8 preventDamage
+	METHOD a spawnFire (I)V
+		ARG 1 extraFireBlocksCount
+	METHOD d setCausingPlayer (Luv;)V
```
Comment (ChloeDawn) [inline mappings/net/minecraft/entity/LightningEntity.mapping]: `setChanneller`

File: mappings/net/minecraft/entity/LightningEntity.mapping [hunk 3]
```diff
@@ -1 +1,16 @@
...
+		ARG 1 causingPlayer
```
Comment (ChloeDawn) [inline mappings/net/minecraft/entity/LightningEntity.mapping]: `channeller`

**Review by ChloeDawn — CHANGES_REQUESTED** [review]
The player does not "cause" the lightning, they channel the lightning. They are the `channeller` of "channelled lightning"

File: mappings/net/minecraft/entity/LightningEntity.mapping [hunk 4]
```diff
@@ -1 +1,16 @@
 CLASS app net/minecraft/entity/LightningEntity
+	FIELD b randomSeed J
+	FIELD c ticksBeforeDespawn I
+	FIELD d fireSpawnCount I
+	FIELD e preventDamage Z
```
Comment (ChloeDawn) [inline mappings/net/minecraft/entity/LightningEntity.mapping]: `harmless`, maybe? Negative connotation for booleans is bad.

File: mappings/net/minecraft/entity/LightningEntity.mapping [hunk 5]
```diff
@@ -1 +1,16 @@
 CLASS app net/minecraft/entity/LightningEntity
+	FIELD b randomSeed J
+	FIELD c ticksBeforeDespawn I
```
Comment (InsomniaKitten) [inline mappings/net/minecraft/entity/LightningEntity.mapping]: `remainingTicks`?

Comment (ChloeDawn) [inline mappings/net/minecraft/entity/LightningEntity.mapping]: `remainingTicks`?

File: mappings/net/minecraft/entity/LightningEntity.mapping [hunk 6]
```diff
@@ -1 +1,16 @@
 CLASS app net/minecraft/entity/LightningEntity
+	FIELD b randomSeed J
```
Comment (ChloeDawn) [inline mappings/net/minecraft/entity/LightningEntity.mapping]: `seed` should be fine for this

**Review by ChloeDawn — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/LightningEntity.mapping (hunk 5 above)
Comment (asiekierka) [reply mappings/net/minecraft/entity/LightningEntity.mapping]: Might be a bit vague?

File: mappings/net/minecraft/entity/LightningEntity.mapping (hunk 4 above)
Comment (asiekierka) [reply mappings/net/minecraft/entity/LightningEntity.mapping]: "noDamage"?

File: mappings/net/minecraft/entity/LightningEntity.mapping [hunk 7]
```diff
@@ -1,16 +1,16 @@
...
-	FIELD b randomSeed J
-	FIELD c ticksBeforeDespawn I
-	FIELD d fireSpawnCount I
-	FIELD e preventDamage Z
-	FIELD f causingPlayer Luv;
+	FIELD b seed J
+	FIELD c ambientTick I
+	FIELD d remainingActions I
+	FIELD e cosmetic Z
+	FIELD f channeller Luv;
```
Comment (asiekierka) [inline mappings/net/minecraft/entity/LightningEntity.mapping:6]: Okay, those names are an improvement. I'm just a bit curious about the origin of "channeller" - that is, is it something we have invented, or is there precedent?

**Review by asiekierka — APPROVED** [review]

File: mappings/net/minecraft/entity/LightningEntity.mapping (hunk 7 above)
Comment (MisterPeModder) [reply mappings/net/minecraft/entity/LightningEntity.mapping:6]: This field is set when the LightningEntity is spawned by a channelling trident

Comment (MisterPeModder) [reply mappings/net/minecraft/entity/LightningEntity.mapping:6]: @InsomniaKitten came up with this name so idk

File: mappings/net/minecraft/entity/LightningEntity.mapping [hunk 8]
```diff
@@ -1 +1,16 @@
...
+	METHOD <init> (Lbdb;DDDZ)V
+		ARG 1 world
+		ARG 2 x
+		ARG 4 y
+		ARG 6 z
+		ARG 8 cosmetic
+	METHOD a spawnFire (I)V
+		ARG 1 extraFireBlocksCount
```
Comment (ChloeDawn) [inline mappings/net/minecraft/entity/LightningEntity.mapping]: `spreadAttempts`

**Review by ChloeDawn — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/LightningEntity.mapping (hunk 8 above)
Comment (MisterPeModder) [reply mappings/net/minecraft/entity/LightningEntity.mapping]: `spreadAttempts` is a way better name IMO

**Review by ChloeDawn — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**